### PR TITLE
Add validation messages for `minChars` and `maxChars` fixes #24

### LINKED ISSUE #24: Add validation message for min and max tag length
Thanks for maintaining this plugin!  It's great, I've been using it for years.

I have a few sorta related issues:
1. The `minChars` option works correctly but gives the user no feedback explaining why their tag was rejected. It seems obviously bad UI to enforce a validation without giving the user some sort of feedback.
2. I would like a similar option for `maxChars`. I know it sounds ridiculous but our users sometimes paste in a long list of keywords, hit tab, and the plugin essentially creates one giant keyword. This seems like a generic enough problem that it should be solved at the library level. Similar to 1, the `maxChars` option should display some type of validation message to the user.

As far as displaying the validation messages, I was thinking we could just dynamically add/remove an `span` which would contain the error message. We could make the `span` class name configurable. That way the plugin could handle the error logic but the client app could still control the styles.

As for the error message text, I was thinking:

```
must be at least X characters
must be fewer than X characters
```

We could make the error text configurable as well; however, I think that might be overkill.

I'm totally willing to make these changes myself and submit a PR. I just want to make sure you're open to these ideas before doing the work.

### PR DESCRIPTION
**Pull request by coryschires**
- Adds a new configuration option for `maxChars`. Works similarly to `minChars`
- Added validation message for `minChars`
- Added validation message for `maxChars`
- Added `onRenderErrorMessage` callback which can be overridden in order to
  customize validation behavior.
- Added corresponding tests.
- Updated readme and change log.
- Bumped version.

### CODE REVIEW

**Comment by knalli** [comment]
Yeah, thank you!

I have some issues
- there is no message in your demo when I have typed in only 2 chars.
- The result list will be shown every time a "suggest" will be tried by aborted because the maximum was reached already (11, 12, 13, ...)

Let's make this PR final, but for the next time consider to make a topic branch for such stuff. It's better :)

Do not let fool by Travis.. the tests are still okay!

File: README.md [hunk 1]
```diff
@@ -325,6 +330,16 @@ afterSelectionRemove: function(element, item, currentSelections){
 }
 ```
 
+#### onRenderErrorMessage (validationData, element, options)
+A callback that is run when a validation message is about to be rendered. `validationData` is an object which contains details about the error which can be helpful when customizing validations. Most commonly, you would override this callback in order to customize a validation behavior.
+
+Example:
+```javascript
+onRenderErrorMessage: function(validationData, element, options){
+  element.after("<span id="+ validationData.id +">Too few characters</span>");
```
Comment (knalli) [inline README.md:339]: Where is the remove callback?

**Comment by coryschires** [comment]
You raise some good points:
1. I totally forgot about the `onRenderErrorMessage` callback, though I do think it's a good idea. I'll add it.
2. Regarding "no message in your demo when I have typed in only 2 chars". I had coded it that way initially but it felt a bit weird having the validation message flashing at me before I'd finished typing my tag name. For example, if I'm adding the tag "delicious", I shouldn't see a validation message when I've typed "de" but have no intention of adding the "de" tag. So I think it's better to only show the `minChars` validation when the user tries to add a tag with too few characters. For `maxChars`, however, I think we want to let the user know as soon as they've typed too many characters – even before they've attempted to add the tag. Does that make sense?
3. Regarding "The result list will be shown every time a "suggest" will be tried by aborted because the maximum was reached already (11, 12, 13, ...)". I don't know what you mean. Can you explain in more detail?

I plan on making these changes tomorrow. Thanks again for being so responsive!

**Comment by knalli** [comment]
Even if the maximum of chars is reached, the box "No results were found" will be flashed. Additionally, the error message will be shown.

**Comment by coryschires** [comment]
@knalli I think those two most recent commits address your concerns. 

Lemme know if it's good to go.

**Comment by knalli** [comment]
Yes, seems to be good. I'll look for it tomorrow's eve.

**Comment by knalli** [comment]
Thank you! (Sorry for the unexpected delay).

**Pull request merged**